src/client.ts

```diff
@@ -184,7 +184,11 @@ export class Client {
     return res.ok
   }
 
-  public async declareConsumer(params: DeclareConsumerParams, handle: ConsumerFunc): Promise<Consumer> {
+  public async declareConsumer(
+    params: DeclareConsumerParams,
+    handle: ConsumerFunc,
+    superStreamConsumer?: SuperStreamConsumer
```

Comment:
Couldn't we simply pass the name contained in the `superStream` property  instead of the whole object?

Reply:
Yes, for sure. The concern I had was that declareConsumer() is a public-facing api and there's not much hint that you shouldn't pass in a super steam name if you don't have a legitimate super stream.

My first thought was to add an internal version of `declareConsumer()` that would be called by `SuperStreamConsumer` only and not be publically accessible to make incorrect usage impossible - however, it's a bit tricky to do and I didn't like the complexity it added.

Instead I chose this simpler way of indicating you should have a valid super stream to call it that way.

This is one of the choices I was hoping for a steer on, since I'm not sure what the best fit would be for this project. If you prefer `superStreamName: string` here, I'd be happy to make that change.

Reply:
Your idea is actually very valid, I did not think of that. I'm still not a superfan of adding a different object to the declareConsumer interface - even as an optional. But I started trying out coding a solution to this and I quickly reached the same conclusion you did: this would add a significant amount of extra complexity and at this stage it is simply not worth it. So I can only agree with your findings, good work!